### PR TITLE
Build fixes for older compilers

### PR DESCRIPTION
**Pull request by tofurky**
As was reported in https://github.com/aarond10/https_dns_proxy/issues/54.

Signed-off-by: Matt Merhar <mattmerhar@protonmail.com>

### CODE REVIEW

File: src/utils.c [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 #include <arpa/inet.h>
 #include <stdio.h>
+#include <string.h>
```
Comment (pexcn) [inline src/utils.c]: It seems `include <string.h>` is unnecessary.

Comment (tofurky) [reply src/utils.c]:
```
(jessie.amd64)matt@aquos:~/devel/openwrt/https_dns_proxy/build$ make -j4
Scanning dependencies of target nxjson
[ 11%] Building C object lib/CMakeFiles/nxjson.dir/nxjson/nxjson.c.o
Linking C static library libnxjson.a
[ 11%] Built target nxjson
Scanning dependencies of target https_dns_proxy
[ 33%] [ 33%] [ 44%] [ 55%] Building C object CMakeFiles/https_dns_proxy.dir/src/https_client.c.o
Building C object CMakeFiles/https_dns_proxy.dir/src/dns_poller.c.o
Building C object CMakeFiles/https_dns_proxy.dir/src/logging.c.o
Building C object CMakeFiles/https_dns_proxy.dir/src/dns_server.c.o
[ 66%] Building C object CMakeFiles/https_dns_proxy.dir/src/json_to_dns.c.o
[ 77%] [ 88%] Building C object CMakeFiles/https_dns_proxy.dir/src/main.c.o
Building C object CMakeFiles/https_dns_proxy.dir/src/options.c.o
[100%] Building C object CMakeFiles/https_dns_proxy.dir/src/utils.c.o
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c: In function 'dn_write_name':
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:19:3: warning: incompatible implicit declaration of built-in function 'memcpy'
   memcpy(out, name, namelen);
   ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c: In function 'dn_match':
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:39:9: warning: incompatible implicit declaration of built-in function 'strlen'
     if (strlen(str) < len) {
         ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c: In function 'dn_find_dnptr':
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:74:29: warning: incompatible implicit declaration of built-in function 'strlen'
   const char *nend = name + strlen(name);
                             ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c: In function 'dn_name_nocompress':
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:138:21: warning: incompatible implicit declaration of built-in function 'strlen'
   size_t name_len = strlen(name);
                     ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:140:26: warning: initialization makes pointer from integer without a cast
   char *name_component = strtok_r(name, ".", &savedptr);
                          ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:150:5: warning: incompatible implicit declaration of built-in function 'memcpy'
     memcpy(pos, name_component, l);
     ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:152:20: warning: assignment makes pointer from integer without a cast
     name_component = strtok_r(NULL, ".", &savedptr);
                    ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c: In function 'unescape':
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:165:23: warning: incompatible implicit declaration of built-in function 'strlen'
   const char *e = s + strlen(in);
                       ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:172:5: warning: incompatible implicit declaration of built-in function 'strncpy'
     strncpy(o, s, *olen);
     ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c: In function 'b32hexdec':
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:320:23: warning: incompatible implicit declaration of built-in function 'strlen'
   const char *e = s + strlen(buf);
                       ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c: In function 'b64dec':
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:360:13: warning: incompatible implicit declaration of built-in function 'strlen'
   int len = strlen(buf);
             ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c: In function 'hexdec':
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:392:13: warning: incompatible implicit declaration of built-in function 'strlen'
   int len = strlen(buf);
             ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c: In function 'type_bitmap_dec':
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:427:22: warning: initialization makes pointer from integer without a cast
   char *rrtype_str = strtok_r(buf, " ", &saveptr);
                      ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:430:3: warning: incompatible implicit declaration of built-in function 'memset'
   memset(&bits[0], 0, sizeof(bits));
   ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:442:16: warning: assignment makes pointer from integer without a cast
     rrtype_str = strtok_r(NULL, " ", &saveptr);
                ^
/home/matt/devel/openwrt/https_dns_proxy/src/utils.c:453:5: warning: incompatible implicit declaration of built-in function 'memcpy'
     memcpy(pos, &bits[i * 256 / 8], window_len[i]);
     ^
Linking C executable https_dns_proxy
[100%] Built target https_dns_proxy
```

Comment (tofurky) [reply src/utils.c]: maybe not strictly necessary but a terminal full of warnings is no good ;)

**Comment by aarond10** [comment]
Thanks!

**Pull request merged**